h/h_api/bulk_api/model/command.py

```diff
@@ -0,0 +1,133 @@
+from functools import lru_cache
+
+from h.h_api.bulk_api.model.config_body import Configuration
+from h.h_api.bulk_api.model.data_body import (
+    CreateGroupMembership,
+    UpsertGroup,
+    UpsertUser,
+)
+from h.h_api.enums import CommandType, DataType
+from h.h_api.model.base import Model
+from h.h_api.schema import Schema
+
+
+class Command(Model):
+    """A single abstract command provided to the API."""
+
+    validator = Schema.get_validator("bulk_api/wrapper.json")
+    validation_error_title = "Cannot interpret command as the wrapper is malformed"
+
+    def validate(self):
+        super().validate()
+        if isinstance(self.body, Model):
+            self.body.validate()
+
+    @classmethod
+    def create(cls, _type, body):
+        """
+        Create a command.
+
+        :param _type: The CommandType of the command
+        :param body: The payload for the command
+        :return: An instance of Command
+        """
+        return cls([CommandType(_type).value, cls.extract_raw(body)])
+
+    @property
+    def type(self):
+        """
+        Get the command type.
+
+        :return: The CommandType of this command
+        """
+        return CommandType(self.raw[0])
+
+    @property
+    def body(self):
+        """
+        Get the body of this command.
+
+        :return: The raw body
+        """
+        return self.raw[1]
+
+    def __repr__(self):
+        return f"<{self.__class__.__name__} {self.body}>"
```

Comment:
:+1: Always good to have a nice repr.

It's not really important but the Python docs somewhere state that a repr should look like Python code for creating the object. So something like `f"{self.__class__.__name__}({self._type}, {self.body})"`. It's not like it ever gets executed as Python so it doesn't actually matter but it's a nice style. This `< ... >`'s-based style seems to be quite common as well though

Reply:
I was aware of that, but I think this is widely enough broken to not really be a thing.